web/src/hooks/use-list-workflow-runs-api.ts

```diff
@@ -10,23 +10,23 @@ import { HTTPMethod } from "@/types/api";
 const ListWorkflowRunsRequest = z.void();
 const ListWorkflowRunsResponse = z.array(WorkflowRunMetadata);
 
-const buildListWorkflowRunsApi = (workflowId: string) =>
+const buildListWorkflowRunsApi = (workflowId: string, limit?: number) =>
 	api<
 		z.infer<typeof ListWorkflowRunsRequest>,
 		z.infer<typeof ListWorkflowRunsResponse>
 	>({
 		method: HTTPMethod.GET,
-		path: `/api/v1/runs/${workflowId}`,
+		path: `/api/v1/runs/${workflowId}${limit ? `?limit=${limit}` : ""}`,
```

Comment:
this is not how query parameters are supposed to be handled. The `api` builder automatically builds the query parameters for you, you just need to pass the parameters. It'll know what to do because you specify the HTTP method as `GET`.
See `useGetWorkflowApi` as a reference and review `api.ts`.